briefcase/app.py

```diff
@@ -219,14 +222,32 @@ def generate_app_template(self, extra_context=None):
             extra_context=_extra_context
         )
 
-    def git_checkout(self, path):
+    def _has_cookiecutter_json(self, template_path):
+        cookiecutter_json_path = os.path.expanduser('%s/cookiecutter.json' % template_path)
```

Comment:
We don't need the call to `os.path.expanduser` in here, because it was already called before.
This will be better:

    cookiecutter_json_path = os.path.join(template_path, 'cookiecutter.json')

This will reduce the probability of having problems with directories (say, in case the home directory has names or something).